rust/main/hyperlane-base/src/kas_hack/logic_loop.rs

```diff
@@ -136,29 +142,43 @@ where
         logs
     }
 
-    // TODO: not used yet, and why would it be a loop?
-    pub fn run_confirmation_loop(mut self, task_monitor: TaskMonitor) -> JoinHandle<()> {
-        let name = "dymension_kaspa_confirmation_loop";
-        tokio::task::Builder::new()
-            .name(name)
-            .spawn(TaskMonitor::instrument(
-                &task_monitor,
-                async move {
-                    self.confirmation_loop().await;
-                }
-                .instrument(info_span!("Kaspa Monitor")),
-            ))
-            .expect("Failed to spawn kaspa monitor task")
-    }
 
-    async fn confirmation_loop(&mut self) {
-        loop {
-            time::sleep(Duration::from_secs(10)).await;
+    /* -------------------------------------------------------------------------- */
+    /*                                 sync logic                                 */
+    /* -------------------------------------------------------------------------- */
+    // TODO: move to a separate file
+
+    /// Sync relayer that blocks until the system is synced
+    /// Checks if the outpoint committed on the hub is already spent on Kaspa
+    /// If not synced, prepares progress indication and submits to hub
+    pub async fn sync_relayer_if_needed(&self) -> Result<(), Error>{
+        // get anchor utxo from hub
+        let resp = self.hub_mailbox.provider().grpc().outpoint(None).await?;
+        let anchor_utxo = resp.outpoint;
```

Comment:
I've added cosmos grpc client to kaspa provider

https://github.com/dymensionxyz/hyperlane-monorepo/blob/78e07265bb12b9c5f23cc5bdbe9e24f102129792/rust/main/chains/dymension-kaspa/src/providers/provider.rs#L45

so you could also do this logic in provider using that, instead of here
probably neater